### PR TITLE
define for AirbotF4 board (ChibiOS HAL)

### PR DESCRIPTION
**Pull request by anbello**
modified:   libraries/AP_HAL/AP_HAL_Boards.h
	new file:   libraries/AP_HAL_ChibiOS/hwdef/airbotf4/hwdef.dat

### CODE REVIEW

**Comment by anbello** [comment]
Maybe this PR will be of little interest to most people because the AirbotF4 is an old board with little features (no on-board baro, SD, OSD) anyway it is here.

File: libraries/AP_HAL_ChibiOS/hwdef/airbotf4/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+PA9 USART1_TX USART1
+
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+#PA15 OSD_CS CS
+#PB3 BMP280_CS CS
```
Comment (sh83) [inline libraries/AP_HAL_ChibiOS/hwdef/airbotf4/hwdef.dat]: SPI3 is used to connect winbond spi flash.
It currently not supported, but I think it should be enableb.
Something like
PB3 FLASH_CS CS
....

**Comment by night-ghost** [comment]
Cool! But it seems that I should to stop publishing the changes, because it seems that everything goes to the fact that all my work will be transferred by thankful users to ChibiOs, after which my HAL will be thrown out of the master like it was with FlyMaple.

 So is not it better to do this in advance?

**Comment by sh83** [comment]
@night_ghost I think it is great to have 2 independent fw. At least(but not limited to) to debug and compare purposes.

**Comment by night-ghost** [comment]
>I think it is great to have 2 independent fw(hal)

sure, but didn't you see how my solutions goes used without any credits? How  it can be compared while my PRs slill hangs for a while - where ChibiOS PRs are merged immidiately, without any discussion? My HAL had been  many times was broken by another PRs, without any appologises - but I don't have so much time to fight with the Ardupilot team alone.

**Comment by sh83** [comment]
@night_ghost I think anbello will remove this pr if you like.

**Comment by night-ghost** [comment]
please don't, I have the better choice

**Comment by anbello** [comment]
@nigth_ghost I did it only because it was really easy to do it, starting
from omnibusf4pro hwdef.dat only few changes, if you prefer I can retire it

Il dom 6 mag 2018, 18:14 night-ghost <notifications@github.com> ha scritto:

**Comment by tridge** [comment]
@night-ghost I think you have some errors in your comments:
1) flymaple HAL was removed in September 2015 with the support and agreement of the developer of that HAL, Mike McCauley, when he no longer used it himself and didn't have the time to maintain it. 
2) I have been prioritizing my reviewing efforts on F4Light for quite some time, working on getting your PRs merged ahead of anything else. I had hoped you would notice and appreciate the effort

**Comment by anbello** [comment]
I added anther commit with mods suggested by @sh83, maybe I had to modify the existent commit and not add another but I don't know how, it is the first PR for me and I know little to nothing of git.

**Comment by anbello** [comment]
It is right, I have flown with it on an f450 quad

Il lun 7 mag 2018, 09:22 sh83 <notifications@github.com> ha scritto:

**Pull request merged**